temporal-sdk/src/main/java/io/temporal/internal/common/SdkFlag.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2022 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ * Copyright (C) 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this material except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.internal.common;
+
+/**
+ * SdkFlag represents a flag used to help version the sdk internally to make breaking changes in
+ * workflow logic.
+ */
+public enum SdkFlag {
+  UNSET(0),
+  /*
+   * Changes behaviour of GetVersion to not yield if no previous call existed in history.
+   */
+  SKIP_YIELD_ON_DEFAULT_VERSION(1),
+  UNKNOWN(Integer.MAX_VALUE);
```

Comment:
Why have an unknown flag? Just make the return from `GetValue` (sic) as `Optional`

Reply:
`UNKNOWN ` is not the same as an empty optional

Reply:
What I'm saying is it shouldn't be the same. You should only have enumerates for real flags IMO. There's no reason to have all but one enumerate be a real flag IMO.

Reply:
Eek. I hope we at least don't in TS, Python, and .NET. All those languages have a good way of representing an absent enum without having to work around that some values of the enum are representable in history and some aren't.

Reply:
I don't agree, but y'all's design choice here. At least it's internal.